neo/rawio/neuralynxrawio/neuralynxrawio.py

```diff
@@ -132,26 +133,56 @@ class NeuralynxRawIO(BaseRawIO):
     ]
 
     def __init__(
-        self, dirname="", filename="", exclude_filename=None, keep_original_times=False, strict_gap_mode=True, **kargs
+            self,
+            dirname="",
+            filename="",
+            include_filename=None,
+            exclude_filename=None,
+            keep_original_times=False,
+            strict_gap_mode=True,
+            **kargs
     ):
 
+        if (include_filename is not None) and (filename != ""):
+            raise ValueError("filename and include_filenames cannot be both assigned")
+        if (include_filename is None) and (filename == "") and (dirname == ""):
+            raise ValueError("One of dirname or filename or include_files must be provided.")
+
+        if dirname != "" and (include_filename is not None):
+            include_filename = [os.path.join(dirname, f) for f in include_filename]
+            dirname = ""
+
+        if not isinstance(include_filename, (list, set, np.ndarray)):
+            include_filename = [include_filename]
+        if not isinstance(exclude_filename, (list, set, np.ndarray)):
+            exclude_filename = [exclude_filename]
+
         if dirname != "":
-            self.dirname = dirname
             self.rawmode = "one-dir"
         elif filename != "":
-            self.filename = filename
             self.rawmode = "one-file"
         else:
-            raise ValueError("One of dirname or filename must be provided.")
+            self.rawmode = "multiple-files"
 
+        self.dirname = dirname
+        self.filename = filename
+        self.include_files = include_filename
+        self.exclude_filename = exclude_filename
         self.keep_original_times = keep_original_times
         self.strict_gap_mode = strict_gap_mode
-        self.exclude_filename = exclude_filename
         BaseRawIO.__init__(self, **kargs)
 
     def _source_name(self):
         if self.rawmode == "one-file":
             return self.filename
+        elif self.rawmode == "multiple-files":
+            dirname = {os.path.dirname(x) for x in self.filename}
+            if len(dirname) > 1:
```

Comment:
this should not be possible.... because we make the include_file relative to the dirname so they have to all have the same dirname. In this case if you fix the boolean check above then all you would need to do is:

```python
if self.rawmode == 'one-file':
   return self.filename
else:
    return self.dirname
```

Reply:
Thanks for pointing this out! I meant to check the `dirname` in `include_filename` (forgot to update this after it is added). I was thinking the user may put files in different directories in `include_filename`. Say ['dirname/exp1/file1', 'dirname/exp2/file2'].

Reply:
Neo doesn't really support multi-dir in that way. There needs to be a root dir or a root file for each IO. There could be subfolders from the root, that could be explored, but not completely different root directories. So the source should return either a dirname or a filename.

Reply:
OK, will just extract `dirname` from the first file path, assuming all files are in the same directory.